src/vmm/src/vstate/vcpu/mod.rs

```diff
@@ -110,6 +113,10 @@ pub struct Vcpu {
     response_receiver: Option<Receiver<VcpuResponse>>,
     // The transmitting end of the responses channel owned by the vcpu side.
     response_sender: Sender<VcpuResponse>,
+
+    // Exit reason used to test run_emulation function.
+    #[cfg(test)]
+    vcpu_exit_reason: Mutex<Option<std::result::Result<VcpuExit<'static>, errno::Error>>>,
```

Comment:
Since this is not a global anymore, it doesn't have to be a `Mutex` anymore.

```suggestion
    vcpu_test_exit_reason: Option<std::result::Result<VcpuExit<'static>, errno::Error>>,
```

Reply:
I have tried to implement Clone trait but you can not implement a trait for a struct defined in another crate.
I have also tried to use RefCell, but won't compile because moving a RefCell between threads is not considered safe.
https://github.com/firecracker-microvm/firecracker/blob/364cf32bbe00e511d25631a37a0e342186309d21/src/vmm/src/vstate/vcpu/mod.rs#L233
https://github.com/firecracker-microvm/firecracker/blob/364cf32bbe00e511d25631a37a0e342186309d21/src/vmm/src/vstate/vcpu/mod.rs#L825